src/components/PostDetails.tsx

```diff
@@ -157,9 +94,7 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
           </button>
         )}
 
-        {loaded && !hasError && visible && (
-          <NewCommentForm onSubmit={addComment} />
-        )}
+        {!isLoading && !error && visible && <NewCommentForm />}
```

Comment:
Same issue here. The `NewCommentForm` will never be displayed because `isLoading` and `visible` will not be true at the same time. You should separate these conditions.